nabs/plans.py

```diff
@@ -0,0 +1,601 @@
+
+    This is a :func:`duration_scan` in one dimension that also runs the DAQ,
+    sets the stage velocity to match the configured sweep time, and returns
+    the motor to the starting position at the end of the scan.
+
+    Parameters
+    ----------
+    time_motor : DelayNewport
+        The movable device in egu seconds.
+
+    time_points : list of float
+        The times in second to move between.
+
+    sweep_time : float
+        The duration we take to move from one end of the range to the other.
+
+    duration : float, optional
+        If provided, the time to run in seconds. If omitted, we'll run forever.
+
+    record : bool, optional
+        Whether or not to record the run in the DAQ. Defaults to True because
+        we don't want to accidentally skip recording good runs.
+
+    See Also
+    --------
+    :func:`delay_scan`
+    :func:`duration_scan`
+    """
+
+    @nbpp.daq_during_decorator(record=record, controls=[time_motor])
+    def inner_daq_delay_scan():
+        return (yield from delay_scan(time_motor, time_points, sweep_time,
+                                      duration=duration))
+
+    return (yield from inner_daq_delay_scan())
+
+
+# The rest of this file contains thin wrappers around bluesky built-ins.
+# These exist to mimic an older hutch python API,
+# easing the transition to bluesky.
+
+
+@nbpp.daq_step_scan_decorator
+def daq_count(detectors=None, num=1, delay=None, *, per_shot=None, md=None):
+    """
+    Take repeated DAQ runs with no motors.
+
+    This is an LCLS-I DAQ version of bluesky's built-in count plan.
+
+    Parameters
+    ----------
+    detectors : list, optional
+        List of 'readable' objects to read at every step.
+
+    num : int, optional
+        Number of readings to take; default is 1.
+        If None, capture data until canceled.
+
+    delay : iterable or scalar, optional
+        Time delay in seconds between successive readings; default is 0.
+
+    events : int, optional
+        Number of events to take at each step. If omitted, uses the
+        duration argument or the last configured value.
+
+    duration : int or float, optional
+        Duration of time to spend at each step. If omitted, uses the events
+        argument or the last configured value.
+
+    record : bool, optional
+        Whether or not to record the run in the DAQ. Defaults to True because
+        we don't want to accidentally skip recording good runs.
+
+    use_l3t : bool, optional
+        Whether or not the use the l3t filter for the events argument. Defaults
+        to False to avoid confusion from unconfigured filters.
+
+    per_shot : callable, optional
+        Hook for customizing action of inner loop (messages per step).
+        Expected signature ::
+
+           def f(detectors: Iterable[OphydObj]) -> Generator[Msg]:
+               ...
+
+    md : dict, optional
+        Additional metadata to include in the start document.
+    """
+
+    if not detectors:
```

Comment:
This branch is because the decorator is set up to do "everything"- assuming you're already reading something else in the plan. There is actually no way for the mutator to know what to do if there aren't any other trigger/read messages in the plan. I have to pass "something" into `count` and thought allowing the daq to be explicitly passed in was the cleanest.

Reply:
I think this made less sense due to the order I reviewed it in. That being said, your reply is a pretty much perfect comment to explain why this is here, if you'd be OK copying it in and annotating the if block.

Reply:
Yep, I will annotate this block